src/Orchard/Environment/DefaultOrchardHost.cs

```diff
@@ -343,6 +348,20 @@ public class DefaultOrchardHost : IOrchardHost, IShellSettingsManagerEventHandle
             _tenantsToRestart.GetState().Add(context.Settings);
         }
 
+        private void BlockRequestsDuringSetup() {
+            if (_shellContexts == null)
+                return;
+
+            // If there's only one tenant and it's initializing, return a Service Unavailable HTTP status code.
+            var shellContexts = _shellContexts.ToList();
+            if (shellContexts.Count == 1 && shellContexts[0].Settings.State == TenantState.Initializing) {
+                var response = _httpContextAccessor.Current().Response;
+                response.StatusCode = 503;
+                response.StatusDescription = "Orchard is currently setting up. Please check back in a few moments.";
+                response.Write("Orchard is currently setting up. Please check back in a few moments.");

```

Comment:
I would suggest a more techno-formal wording:
"This tenant is currently initializing. Please try again later."